File: backend/satisfactory-blueprint/satisfactory-blueprint/src/main/java/com/satisfactory/blueprint/dto/UpdateTransportRouteRequest.java
package com.satisfactory.blueprint.dto;

import com.satisfactory.blueprint.entity.enums.VehicleType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateTransportRouteRequest {
    private Long id;
    private Long planId;
    private String stationName;
    private Double capacity;
    private Long assignedItemId;
    private List<Double> cartCapacities;
    private List<CartAllocationDto> cartAllocations;
}

File: frontend/satisfactory-blueprint-designer/src/types/updateTransportRouteRequest.ts
// src/types/updateTransportRouteRequest.ts
import type { CartAllocationDto } from './cartAllocation';

/**
 * Mirrors com.satisfactory.blueprint.dto.UpdateTransportRouteRequest
 */
export interface UpdateTransportRouteRequest {
  id: number;
  planId: number;
  stationName: string;
  capacity: number;
  assignedItemId: number;
  cartCapacities: number[];
  cartAllocations: CartAllocationDto[];
}
